api.go

```diff
@@ -144,6 +144,14 @@ type Stats struct {
 	// Number of candidate matches as a result of searching ngrams.
 	NgramMatches int
 
+	// Priority of the shard that was searched.
+	ShardPriority float64
+
+	// Maximum priority of a shard that is being searched in parallel. This is used to reorder results
+	// when the result set is known to be stable-- that is, when a result's ShardPriority is greater than
+	// the max(MaxPendingShardPriority) from the latest results of each backend, it can be returned to the user.
+	MaxPendingShardPriority float64
+
```

Comment:
We rely on stats being aggregateable. See for example how we use stats.Add in the stream client. Stats might not be related to a single shard. Maybe it makes more sense to attach this kind of information to the top-level of api.SearchResult?

Reply:
Agreed. All fields on this we should be able to aggregate. Adding something to api.SearchResult sounds good to me. If we can avoid exposing that impl detail to clients that would be ideal, but maybe streaming clients could also use this info?

Reply:
Moved to a Progress struct.